web/src/copy_and_paste.js

```diff
@@ -584,6 +599,12 @@ export function paste_handler(event) {
         ) {
             event.preventDefault();
             event.stopPropagation();
+            if (is_white_space_pre(paste_html)) {
+                // Copied content styled with `white-space: pre` is pasted as is
+                // but formatted as code. We need this for content copied from
+                // VS Code like sources.
+                paste_html = "<pre><code>" + paste_text + "</code></pre>";
```

Comment:
@N-Shar-ma This is an HTML injection bug. **Never insert text into HTML without escaping it.** HTML injection can lead to serious security problems depending on the context, and it’s always a bug regardless of the context.

Reply:
Thank you so much for catching this oversight! I'll try my best to be more mindful about such cases in the future.

The corrected version of the reverted commit is in #29172